File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.c
int waysToSplitArray(int* nums, int numsSize) {
    long long left_sum = nums[0];
    long long right_sum = 0;
    long long total = 0;

    int count = 0;
    int n = numsSize;

    for (int i = 0; i < n; i++) {
        total += nums[i];
    }

    for (int i = 0; i < n - 1; i++) {
        right_sum = total - left_sum;
        if (left_sum >= right_sum) {
            count += 1;
        }
        left_sum += nums[i + 1];
    }
    return count;
}

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        left_sum = nums[0]
        right_sum = 0
        total = 0
        
        count = 0
        n = len(nums)

        for i in range(n):
            total += nums[i]

        for i in range(n-1):
            right_sum = total - left_sum
            if(left_sum >= right_sum):
                count+=1
            left_sum += nums[i+1]
        return count